### DISCUSSION TITLE
How can I tell if a transaction has pending operations?

### DISCUSSION THREAD

**Discussion by RallyTronics** [erthink/libmdbx#352] [Q&A] [answered]
How can I tell if a transaction has pending operations that would be committed if I were to call mdbx_txn_commit?

**Comment by erthink** [comment, marked as answer]
1. Call the [`mdbx_txn_flags()`](https://libmdbx.dqdkfa.ru/group__c__transactions.html#gafd9f6bdac100c247889a79e35152ed39) and test the [`MDBX_TXN_DIRTY`](https://libmdbx.dqdkfa.ru/group__c__transactions.html#gga6ddf4a553ef12632520817465f53d0d6ad889b6c7028afed8f72e949f85c22a20) bit.
2. Use [`mdbx_dbi_flags_ex()`](https://libmdbx.dqdkfa.ru/group__c__statinfo.html#gaec553fcc6b62052840f60979bb30a576) for each opened DBI-handle and examine `state` for `MDBX_TBL_DIRTY` bit. If it is set for at least one DBI-handle then the current transaction have some changes to commit.

**Comment by RallyTronics** [comment]
Thanks!
